src/cloudRun.ts

```diff
@@ -299,4 +303,58 @@ export class CloudRun {
       this.methodOptions,
     );
   }
+
+  async pollService(serviceResponse: run_v1.Schema$Service): Promise<string> {
+    let url = getUrl(serviceResponse);
+    while (!getReadyStatus(serviceResponse) && !url) {
+      await sleep(5000);
+      serviceResponse = await this.getService(serviceResponse.metadata!.name!);
+      url = getUrl(serviceResponse);
+    }
+
+    return url;
+  }
+}
+
+function getReadyStatus(serviceResponse: run_v1.Schema$Service): boolean {
+  const revisionName = get(serviceResponse, 'spec.template.metadata.name');
+  const createdRevision = get(
+    serviceResponse,
+    'status.latestCreatedRevisionName',
+  );
+  const latestRevision = get(serviceResponse, 'status.latestReadyRevisionName');
+  if (revisionName) {
```

Comment:
Comments on the logic when it evaluates to true would be useful to understand what are the ready states.